### DISCUSSION TITLE
is this resolved or someone can help me on this ? > In my case, its a `.crt` file which has bunch of certificates in it from Godaddy.

### DISCUSSION THREAD

**Discussion by devanuj97** [axios/axios#5858] [Q&A]
> In my case, its a `.crt` file which has bunch of certificates in it from Godaddy.
> when i try to do follow the example code @nickuraltsev mentioned, i get following error.
> 
> `{ [Error: unable to get local issuer certificate] code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY' }`

> In my case, its a `.crt` file which has bunch of certificates in it from Godaddy.
> when i try to do follow the example code @nickuraltsev mentioned, i get following error.
> 
> `{ [Error: unable to get local issuer certificate] code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY' }`

I also encountered this problem, can you tell me how you solved it?

_Originally posted by @LNyears in https://github.com/axios/axios/issues/284#issuecomment-493908858_

**Comment by jasonsaayman** [comment]
That error comes from Node's TLS validation, not Axios directly. It means Node cannot build a trusted certificate chain from the server certificate to a trusted root CA.

A few things to check:

- The server should send the full certificate chain, including intermediate certificates.
- The `ca` option should contain the CA/intermediate certificates you want Node to trust, not usually the server certificate itself.
- If your `.crt` file contains multiple certificates, make sure it is valid PEM text with each certificate wrapped like this:

```pem
-----BEGIN CERTIFICATE-----
...
-----END CERTIFICATE-----
```

Example in Node:

```js
import fs from 'node:fs';
import https from 'node:https';
import axios from 'axios';

const httpsAgent = new https.Agent({
  ca: fs.readFileSync('./godaddy-chain.crt')
});

await axios.get('https://example.com', { httpsAgent });
```

Avoid `rejectUnauthorized: false`; it disables certificate validation and should not be used as a real fix.
